lib/paper_trail/has_paper_trail.rb

```diff
@@ -526,7 +526,7 @@ def save_associations_has_and_belongs_to_many(version)
             self.class.paper_trail_save_join_tables.include?(a.name) ||
                 a.klass.paper_trail_enabled_for_model?
           assoc_version_args = {
-            version_id: version.id,
+            version_id: version.transaction_id,
```

Comment:
Remind me again why we store `transaction_id` in the `version_id` column? Is it because there is no "through" table in a HABTM?